[cpp] GCC 4.8: Build failure

**Build Environment**: gcc 4.8.1
**Steps**: from libphonenumber/cpp
$ mkdir build
$ cd build
$ cmake ..
$ make
**Output**:snipped successful output
...
Scanning dependencies of target phonenumber_testing
[  2%] Building CXX object CMakeFiles/phonenumber_testing.dir/src/phonenumbers/asyoutypeformatter.cc.o
In file included from libphonenumber/cpp/src/phonenumbers/asyoutypeformatter.h:39:0,
                 from libphonenumber/cpp/src/phonenumbers/asyoutypeformatter.cc:15:
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h: In constructor ‘i18n::phonenumbers::DefaultDeleter<T>::DefaultDeleter(const i18n::phonenumbers::DefaultDeleter<U>&)’:
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h:52:20: error: typedef ‘U_ptr_must_implicitly_convert_to_T_ptr’ locally defined but not used [-Werror=unused-local-typedefs]
                    U_ptr_must_implicitly_convert_to_T_ptr);
                    ^
libphonenumber/cpp/src/phonenumbers/base/basictypes.h:201:39: note: in definition of macro ‘COMPILE_ASSERT’
   typedef CompileAssert<(bool(expr))> msg[bool(expr) ? 1 : -1]
                                       ^
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h: In constructor ‘i18n::phonenumbers::scoped_ptr<T, D>::scoped_ptr(i18n::phonenumbers::scoped_ptr<U, V>)’:
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h:242:41: error: typedef ‘U_cannot_be_an_array’ locally defined but not used [-Werror=unused-local-typedefs]
     COMPILE_ASSERT(!is_array<U>::value, U_cannot_be_an_array);
                                         ^
libphonenumber/cpp/src/phonenumbers/base/basictypes.h:201:39: note: in definition of macro ‘COMPILE_ASSERT’
   typedef CompileAssert<(bool(expr))> msg[bool(expr) ? 1 : -1]
                                       ^
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h: In member function ‘i18n::phonenumbers::scoped_ptr<T, D>& i18n::phonenumbers::scoped_ptr<T, D>::operator=(i18n::phonenumbers::scoped_ptr<U, V>)’:
libphonenumber/cpp/src/phonenumbers/base/memory/scoped_ptr.h:257:41: error: typedef ‘U_cannot_be_an_array’ locally defined but not used [-Werror=unused-local-typedefs]
     COMPILE_ASSERT(!is_array<U>::value, U_cannot_be_an_array);
                                         ^
libphonenumber/cpp/src/phonenumbers/base/basictypes.h:201:39: note: in definition of macro ‘COMPILE_ASSERT’
   typedef CompileAssert<(bool(expr))> msg[bool(expr) ? 1 : -1]
                                       ^
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/phonenumber_testing.dir/src/phonenumbers/asyoutypeformatter.cc.o] Error 1
make[1]: *** [CMakeFiles/phonenumber_testing.dir/all] Error 2
make: *** [all] Error 2
**Workaround**: in cpp/CMakeLists.txt
:%s/add_definitions ("-Wall -Werror")/add_definitions ("-Wall -Werror -Wno-unused-local-typedefs")
**Fix**: Pending

--- a/cpp/src/phonenumbers/base/basictypes.h
+++ b/cpp/src/phonenumbers/base/basictypes.h
@@ -205,7 +205,7 @@ struct CompileAssert {
 // is important for some other reason.)
 // Use like:
 //   int x ALLOW_UNUSED = ...;
-#if defined(COMPILER_GCC)
+#if defined(__GNUC__)
 #define ALLOW_UNUSED __attribute__((unused))
 #else
 #define ALLOW_UNUSED

--- a/tools/cpp/src/base/basictypes.h
+++ b/tools/cpp/src/base/basictypes.h
@@ -246,7 +246,7 @@ struct CompileAssert {
 // is important for some other reason.)
 // Use like:
 //   int x ALLOW_UNUSED = ...;
-#if defined(COMPILER_GCC)
+#if defined(__GNUC__)
 #define ALLOW_UNUSED __attribute__((unused))
 #else
 #define ALLOW_UNUSED

